server/shared/src/main/scala/org/http4s/server/middleware/CORS.scala

```diff
@@ -320,7 +84,9 @@ sealed class CORSPolicy(
 ) {
   import CORSPolicy._
 
-  def apply[F[_]: Applicative, G[_]](http: Http[F, G]): Http[F, G] = {
+  def apply[F[_]: Applicative, G[_]: Applicative: LoggerFactory](
+      http: Http[F, G]
+  ): G[Http[F, G]] = {
```

Comment:
This is where we get into trouble. To log in this method (see below) we have to return the result in an effect now. Maybe this is ok, but it caused extensive refactoring of the CORS test suite.